kubernetes/resource_kubernetes_service_test.go

```diff
@@ -533,6 +524,94 @@ func TestAccKubernetesService_generatedName(t *testing.T) {
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.0.port", "8080"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.0.protocol", "TCP"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.0.target_port", "80"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.session_affinity", "None"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.type", "ClusterIP"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.publish_not_ready_addresses", "false"),
+					testAccCheckServicePorts(&conf1, []api.ServicePort{
+						{
+							Port:       int32(8080),
+							Protocol:   api.ProtocolTCP,
+							TargetPort: intstr.FromInt(80),
+						},
+					}),
+				),
+			},
+			{
+				Config: requiredProviders() + testAccKubernetesServiceConfig_regression("kubernetes-local", name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesServiceExists("kubernetes_service.test", &conf2),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "metadata.0.name", name),
+					resource.TestCheckResourceAttrSet("kubernetes_service.test", "metadata.0.generation"),
+					resource.TestCheckResourceAttrSet("kubernetes_service.test", "metadata.0.resource_version"),
+					resource.TestCheckResourceAttrSet("kubernetes_service.test", "metadata.0.self_link"),
+					resource.TestCheckResourceAttrSet("kubernetes_service.test", "metadata.0.uid"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.#", "1"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.#", "1"),
+					resource.TestCheckResourceAttrSet("kubernetes_service.test", "spec.0.cluster_ip"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.0.name", ""),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.0.node_port", "0"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.0.port", "8080"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.0.protocol", "TCP"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.port.0.target_port", "80"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.session_affinity", "None"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.type", "ClusterIP"),
+					resource.TestCheckResourceAttr("kubernetes_service.test", "spec.0.publish_not_ready_addresses", "false"),
+					testAccCheckKubernetesServiceForceNew(&conf1, &conf2, false),
+					testAccCheckServicePorts(&conf2, []api.ServicePort{
+						{
+							Port:       int32(8080),
+							Protocol:   api.ProtocolTCP,
+							TargetPort: intstr.FromInt(80),
+						},
+					}),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckKubernetesServiceForceNew(old, new *api.Service, wantNew bool) resource.TestCheckFunc {
```

Comment:
If we're only checking the UID from the metadata I'd bet we could write a generic function for this using the dynamic client so we don't need to write this function for every resource. We can do that in another PR though.